google-cloud-logging/lib/google/cloud/logging/project.rb

```diff
@@ -332,11 +332,12 @@ def async_writer max_queue_size: AsyncWriter::DEFAULT_MAX_QUEUE_SIZE
         #   resource to be associated with written log entries.
         # @param [Hash] labels A set of user-defined data to be associated with
         #   written log entries.
-        # @param [Boolean|AsyncWriter] async_writer An AsyncWriter for the
-        #   logger to transmit log entries. You may also pass true to request
-        #   a new AsyncWriter for this logger, or false to request no
-        #   AsyncWriter (which will cause the logger to make blocking calls).
-        #   Default is true.
+        # @param [Boolean] async Whether entries are written asynchronously.
+        #   When `true` the logger will use use a new AsyncWriter instance. When
+        #   `false` it will write entries synchronously using
+        #   Project#write_entries. Default is true.
+        # @param [AsyncWriter|#write_entries] writer The object to write entries
```

Comment:
I tend to think arguments are less confusing when they're orthogonal. e.g. With this change, the async argument has an effect only if writer is not set. I get that there are trade-offs, and if the preferred style for the libraries is to prefer arguments with a single type rather than preferring orthogonal arguments, then that's fine. But I want to consider a single argument of type [#write_entries|Symbol] or something like that, where the value is either a polymorphic writer or one of the values `:async` or `:sync`.